app/controllers/facility_statements_controller.rb

```diff
@@ -25,61 +25,38 @@ def index
   # GET /facilities/:facility_id/statements/new
   def new_with_search
     @order_details = @order_details.need_statement(@facility)
-    @order_detail_action = :send_statements
+    @order_detail_action = :create
     set_default_start_date if SettingsHelper.feature_on?(:set_statement_search_start_date)
     @layout = "two_column_head"
   end
 
-  # POST /facilities/:facility_id/statements/send_statements
-  def send_statements
-    if params[:order_detail_ids].nil? || params[:order_detail_ids].empty?
-      flash[:error] = I18n.t "controllers.facility_statements.send_statements.no_selection"
-      redirect_to action: :new
-      return
-    end
-    @errors = []
-    to_statement = {}
-    OrderDetail.transaction do
-      params[:order_detail_ids].each do |order_detail_id|
-        od = nil
-        begin
-          ods = OrderDetail.need_statement(current_facility)
-          od = ods.readonly(false).find(order_detail_id)
-          to_statement[od.account] ||= []
-          to_statement[od.account] << od
-        rescue => e
-          @errors << I18n.t("controllers.facility_statements.send_statements.order_error", order_detail_id: order_detail_id)
-        end
-      end
-
-      @account_statements = {}
-      to_statement.each do |account, order_details|
-        statement = Statement.create!(facility: order_details.first.facility, account_id: account.id, created_by: session_user.id)
-        order_details.each do |od|
-          StatementRow.create!(statement_id: statement.id, order_detail_id: od.id)
-          od.statement_id = statement.id
-          @errors << "#{od} #{od.errors}" unless od.save
-        end
-        @account_statements[account] = statement
-      end
-
-      if @errors.any?
-        flash[:error] = I18n.t("controllers.facility_statements.errors_html", errors: @errors.join("<br/>")).html_safe
+  # POST /facilities/:facility_id/statements/create
+  def create
+    @statement_creator = StatementCreator.new(order_detail_ids: params[:order_detail_ids], session_user: session_user, current_facility: current_facility)
+    if @statement_creator.order_detail_ids.nil? || @statement_creator.order_detail_ids.empty?
```

Comment:
Blank!! I was completely BLANK-ing (pun intended) when I was trying to think of the method that covers both :P